core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -1190,6 +1192,9 @@ class PartIdCollector extends SimpleCollector {
   override def collect(doc: Int): Unit = {
     if (partIdDv.advanceExact(doc)) {
       result += partIdDv.longValue().toInt
+      if (result.length >= limit) {
+        throw new CollectionTerminatedException
```

Comment:
So instead of long-running queries ( resulting in taking away resources for large durations > n min ), we are failing fast here ?